src/models/converters/transactions/details.rs

```diff
@@ -28,31 +26,45 @@ impl MultisigTransaction {
                 operation: self.operation.unwrap_or(Operation::CALL),
             }),
             tx_hash: self.transaction_hash.as_ref().map(|hash| hash.to_owned()),
-            detailed_execution_info: Some(DetailedExecutionInfo::Multisig(
-                MultisigExecutionDetails {
-                    submitted_at: self.submission_date.timestamp_millis(),
-                    nonce: self.nonce,
-                    safe_tx_hash: self.safe_tx_hash.to_owned(),
-                    executor: self.executor.to_owned(),
-                    signers: safe_info.owners,
-                    confirmations_required: self
-                        .confirmations_required
-                        .unwrap_or(safe_info.threshold),
-                    confirmations: self
-                        .confirmations
-                        .as_ref()
-                        .unwrap_or(&vec![])
-                        .into_iter()
-                        .map(|confirmation| MultisigConfirmation {
-                            signer: confirmation.owner.to_owned(),
-                            signature: confirmation.signature.to_owned(),
-                            submitted_at: confirmation.submission_date.timestamp_millis(),
-                        })
-                        .collect(),
-                },
-            )),
+            detailed_execution_info: Some(DetailedExecutionInfo::Multisig(self.build_execution_details(safe_info, gas_token))),
         })
     }
+
+    fn build_execution_details(&self, safe_info: SafeInfo, gas_token: Option<TokenInfo>) -> MultisigExecutionDetails {
+        MultisigExecutionDetails {
+            submitted_at: self.submission_date.timestamp_millis(),
+            nonce: self.nonce,
+            safe_tx_hash: self.safe_tx_hash.to_owned(),
+            executor: self.executor.to_owned(),
+            signers: safe_info.owners,
+            confirmations_required: self
+                .confirmations_required
+                .unwrap_or(safe_info.threshold),
+            confirmations: self
+                .confirmations
+                .as_ref()
+                .unwrap_or(&vec![])
+                .into_iter()
+                .map(|confirmation| MultisigConfirmation {
+                    signer: confirmation.owner.to_owned(),
+                    signature: confirmation.signature.to_owned(),
+                    submitted_at: confirmation.submission_date.timestamp_millis(),
+                })
+                .collect(),
+            signature_info: Some(self.build_signature_info(gas_token)),
+        }
+    }
+
+    fn build_signature_info(&self, gas_token: Option<TokenInfo>) -> SignatureInfo {
+        SignatureInfo {
```

Comment:
Why add another level for this and not put it on the same level as `safe_tx_hash` and `nonce`? Nonce is similar to all the info inside this struct, as it also is only used for signing.

Reply:
I wanted to keep this in isolation to prevent having to modify tests too soon, before I was sure of what was necessary. I will flatten the structure.